### DISCUSSION TITLE
Can Tuyaconvert be used to ascertain if I have a ESP core?

### DISCUSSION THREAD

**Discussion by KDMcMullan** [ct-Open-Source/tuya-convert#1113] [Q&A]
Hi. Trying to use Tuyaconvert, because I've bought a device and I don't know if it's an ESP core and i don't want to crack the case. If it's not an ESP, I'll take it back to the shop for a refund.

Tuyaconvert tells me:
_Your device's firmware is too new.
Tuya patched the PSK vulnerability that we use to establish a connection._

I'm hoping to ascertain how accurate this diagnosis is. If it is genuinely that the firmware is too new, I'll do a serial flash, which is my usual method.

I did consult https://github.com/ct-Open-Source/tuya-convert/wiki/Collaboration-document-for-PSK-Identity-02. I found my smarthack.psklog and it does, in fact have ID 2.

Does that mean I have the ESP? Is it possible to know?

Thanks.

**Comment by kueblc** [comment]
Check the MAC address against an OUI database to see if the vendor is Espressif. You can find the MAC address of your device in the logs if tuya-convert made contact.

**Reply by KDMcMullan** [reply]
Bother. According to https://macvendors.com/, the vendor is "Tuya Smart Inc." I think this is not the answer I was hoping for.

Thanks.
